diff --git a/config.cmake b/config.cmake
index a3d5b61d..783cee69 100644
--- a/config.cmake
+++ b/config.cmake
@@ -86,7 +86,7 @@ if(TTK_BUILD_DOCUMENTATION)
   find_package(Doxygen)
 endif()
 
-find_package(Boost COMPONENTS system)
+find_package(Boost)
 if(NOT Boost_FOUND)
   find_package(Boost REQUIRED)
   if(Boost_FOUND)
diff --git a/core/base/ftmTree/CMakeLists.txt b/core/base/ftmTree/CMakeLists.txt
index 757f4532..f517ea84 100644
--- a/core/base/ftmTree/CMakeLists.txt
+++ b/core/base/ftmTree/CMakeLists.txt
@@ -22,7 +22,6 @@ ttk_add_base_library(ftmTree
     triangulation
     geometry
     Boost::boost
-    Boost::system
     )
 
 option(TTK_ENABLE_FTM_TREE_PROCESS_SPEED "Enable FTM tree process speed" OFF)
diff --git a/core/base/ftrGraph/CMakeLists.txt b/core/base/ftrGraph/CMakeLists.txt
index 8cd2f2ef..4854a818 100644
--- a/core/base/ftrGraph/CMakeLists.txt
+++ b/core/base/ftrGraph/CMakeLists.txt
@@ -49,7 +49,6 @@ ttk_add_base_library(ftrGraph
     triangulation
     scalarFieldCriticalPoints
     Boost::boost
-    Boost::system
     ${profiler_lib}
     )
 
diff --git a/core/base/icosphere/CMakeLists.txt b/core/base/icosphere/CMakeLists.txt
index 2070433c..ed9a63da 100644
--- a/core/base/icosphere/CMakeLists.txt
+++ b/core/base/icosphere/CMakeLists.txt
@@ -6,5 +6,4 @@ ttk_add_base_library(icosphere
   DEPENDS
     triangulation
     Boost::boost
-    Boost::system
   )
diff --git a/core/base/trackingFromOverlap/CMakeLists.txt b/core/base/trackingFromOverlap/CMakeLists.txt
index edb62657..febcdc1b 100644
--- a/core/base/trackingFromOverlap/CMakeLists.txt
+++ b/core/base/trackingFromOverlap/CMakeLists.txt
@@ -6,5 +6,4 @@ ttk_add_base_library(trackingFromOverlap
   DEPENDS
     triangulation
     Boost::boost
-    Boost::system
     )
diff --git a/core/vtk/ttkCinemaQuery/ttk.module b/core/vtk/ttkCinemaQuery/ttk.module
index 4ab12502..c47022b6 100644
--- a/core/vtk/ttkCinemaQuery/ttk.module
+++ b/core/vtk/ttkCinemaQuery/ttk.module
@@ -8,4 +8,3 @@ DEPENDS
   cinemaQuery
   ttkAlgorithm
   Boost::boost
-  Boost::system
diff --git a/core/vtk/ttkCinemaWriter/ttk.module b/core/vtk/ttkCinemaWriter/ttk.module
index cf413906..b29af4f5 100644
--- a/core/vtk/ttkCinemaWriter/ttk.module
+++ b/core/vtk/ttkCinemaWriter/ttk.module
@@ -8,4 +8,3 @@ DEPENDS
   ttkAlgorithm
   ttkTopologicalCompressionWriter
   Boost::boost
-  Boost::system
diff --git a/core/vtk/ttkTrackingFromOverlap/ttk.module b/core/vtk/ttkTrackingFromOverlap/ttk.module
index 09d7d32b..51629061 100644
--- a/core/vtk/ttkTrackingFromOverlap/ttk.module
+++ b/core/vtk/ttkTrackingFromOverlap/ttk.module
@@ -8,4 +8,3 @@ DEPENDS
   trackingFromOverlap
   ttkAlgorithm
   Boost::boost
-  Boost::system
