diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e8d59fb..204786ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,11 +92,11 @@ install(
   EXPORT
     TTKBaseTargets
   RUNTIME DESTINATION
-    ${CMAKE_INSTALL_BINDIR}/ttk
+    ${CMAKE_INSTALL_BINDIR}
   ARCHIVE DESTINATION
-    ${CMAKE_INSTALL_LIBDIR}/ttk
+    ${CMAKE_INSTALL_LIBDIR}
   LIBRARY DESTINATION
-    ${CMAKE_INSTALL_LIBDIR}/ttk
+    ${CMAKE_INSTALL_LIBDIR}
   )
 
 install(
@@ -140,11 +140,11 @@ if(TTK_BUILD_VTK_WRAPPERS)
     EXPORT
       TTKVTKTargets
     RUNTIME DESTINATION
-      ${CMAKE_INSTALL_BINDIR}/ttk
+      ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION
-      ${CMAKE_INSTALL_LIBDIR}/ttk
+      ${CMAKE_INSTALL_LIBDIR}
     LIBRARY DESTINATION
-      ${CMAKE_INSTALL_LIBDIR}/ttk
+      ${CMAKE_INSTALL_LIBDIR}
     )
 
   install(
