diff --git a/CMake/CompilerFlags.cmake b/CMake/CompilerFlags.cmake
index 5919753b..044fe9ee 100644
--- a/CMake/CompilerFlags.cmake
+++ b/CMake/CompilerFlags.cmake
@@ -20,14 +20,6 @@ if (NOT MSVC) # GCC and Clang
     list(APPEND TTK_COMPILER_FLAGS -O0 -g -pg)
   endif()
 
-  # hardened linker flags for Clang and GCC
-  if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-    list(APPEND TTK_LINKER_FLAGS
-      -Wl,--as-needed
-      -Wl,--no-undefined
-      )
-  endif()
-
 elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" AND "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
   # clang-cl on Windows
   # c.f. https://stackoverflow.com/questions/50857779/cmake-detects-clang-cl-as-clang
