diff --git a/CMake/CompilerFlags.cmake b/CMake/CompilerFlags.cmake
index 0c44f5c7..68fa5542 100644
--- a/CMake/CompilerFlags.cmake
+++ b/CMake/CompilerFlags.cmake
@@ -15,13 +15,6 @@ if (NOT MSVC) # GCC and Clang
     list(APPEND TTK_COMPILER_FLAGS -O0 -g -pg)
   endif()
 
-  # hardened linker flags for Clang and GCC
-  if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-    list(APPEND TTK_LINKER_FLAGS
-      -Wl,--as-needed
-      -Wl,--no-undefined
-      )
-  endif()
 
 elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" AND "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
   # clang-cl on Windows
